src/librustc_resolve/resolve_imports.rs

```diff
@@ -763,8 +756,18 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
 
             match path_res {
                 PathResult::Module(module) => module,
-                PathResult::Indeterminate => return false,
-                PathResult::NonModule(..) | PathResult::Failed { .. } => return true,
+                PathResult::Indeterminate => {
+                    // Do not complain about unused import when encountering a bad import.
+                    self.r.used_imports.insert((directive.id, TypeNS));
```

Comment:
This is incorrect, indeterminate resolutions can become successfully resolved later.

Reply:
See https://github.com/rust-lang/rust/pull/64054#discussion_r320361845